centaur/src/main/scala/centaur/api/CentaurCromwellClient.scala

```diff
@@ -56,6 +56,7 @@ object CentaurCromwellClient extends StrictLogging {
         submittedWorkflow =>
           for {
             _ <- IO(logger.info(s"Submitting ${workflow.testName} returned workflow id ${submittedWorkflow.id}"))
+            _ = workflow.addSubmittedWorkflow(submittedWorkflow)
```

Comment:
Good idea to put the ids on the workflow itself!  I'm not sure this quite works, though. I see how we get here through `TestFormulas.runSuccessfulWorkflow`, which is called from `runWorkflowTwiceExpectingCallCaching`. But it's called like this:
```
for {
      ...
      firstWF <- runSuccessfulWorkflow(workflowDefinition)
      secondWf <- runSuccessfulWorkflow(workflowDefinition.secondRun)
      ...
}
```
`workflowDefinition.secondRun` returns a copy of the the `Workflow`, so adding the submitted workflow id to it won't result in the `CentaurTestCase`'s copy of the workflow being updated. Or am I missing something? 

Much less elegant, but I think it would work to add the second and third workflows directly in the `TestFormulas` methods?

Reply:
Great catch! I can try moving the submitted ids to the `CentaurTestCase` instead; not totally sure that's going to work so we may have to fall back to those explicit adds in the twice / thrice methods as you suggest.